HTTP 405 Method Not Allowed when attempting to update slskd port

It seems slskd has changed this endpoint to use PUT instead of POST:
https://github.com/slskd/slskd/commit/74ca21dc0566b081f9fa372d6895b12bc0aaa569#diff-d12e101d6bf0c6975f8fdc540ed2e8f341e97d7ae1afec04b1ee1fbdfd232d0aR172

```
2026-03-07 21:04:00,609 [ERROR] Retryable error in lifecycle
Traceback (most recent call last):
  File "/glueforward/slskd.py", line 151, in set_port
    self.__update_config(config)
  File "/glueforward/slskd.py", line 131, in __update_config
    self.__request(
  File "/glueforward/slskd.py", line 88, in __request
    self._handle_request_exception(
  File "/glueforward/service_client.py", line 43, in _handle_request_exception
    raise exception
  File "/glueforward/slskd.py", line 83, in __request
    response.raise_for_status()
  File "/usr/local/lib/python3.12/site-packages/httpx/_models.py", line 829, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Client error '405 Method Not Allowed' for url 'http://<redacted>/api/v0/options/yaml'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/glueforward/main.py", line 125, in run
    self._loop()
  File "/glueforward/main.py", line 117, in _loop
    self.__service_client.set_port(forwarded_port)
  File "/glueforward/slskd.py", line 156, in set_port
    raise SlskdSetPortFailed("Failed to update port") from e
slskd.SlskdSetPortFailed: ('Failed to set slskd listening port', 'Failed to update port')
2026-03-07 21:04:00,610 [INFO] Retrying in 10 seconds
```

--- a/glueforward/slskd.py
+++ b/glueforward/slskd.py
@@ -129,7 +129,7 @@ class SlskdClient(ServiceClient):
         # The slskd API expects a JSON payload containing the YAML as a string property
         # json.dumps() ensures proper string escaping for the JSON payload
         self.__request(
-            method="post",
+            method="put",
             url="/api/v0/options/yaml",
             data=json.dumps(yaml_content),
             headers={"Content-Type": "application/json"}

